vars/buildDockerAndPublishImage.groovy

```diff
@@ -230,102 +187,149 @@ def call(String imageShortName, Map userConfig=[:]) {
           } // if else
         } // each
 
-        // Automatic tagging on principal branch is not enabled by default, and disabled if disablePublication is set to true
-        if (semVerEnabledOnPrimaryBranch) {
-          stage("Semantic Release of ${imageName}") {
-            echo "Configuring credential.helper"
-            // The credential.helper will execute everything after the '!', here echoing the username, the password and an empty line to be passed to git as credentials when git needs it.
-            if (isUnix()) {
-              sh 'git config --local credential.helper "!set -u; echo username=\\$GIT_USERNAME && echo password=\\$GIT_PASSWORD && echo"'
-            } else {
-              // Using 'bat' here instead of 'powershell' to avoid variable interpolation problem with $
-              bat 'git config --local credential.helper "!sh.exe -c \'set -u; echo username=$GIT_USERNAME && echo password=$GIT_PASSWORD && echo"\''
-            }
+        if (env.BRANCH_IS_PRIMARY) {
+          // Automatic tagging on principal branch is not enabled by default, show potential next version in PR anyway
+          // This stage is now inside the primary branch check
+          if (finalConfig.automaticSemanticVersioning) {
+            stage("Get Next Version of ${imageName}") {
+              String imageInTag = '-' + imageName.replace('-','').replace(':','').toLowerCase()
```

Comment:
After checking, it looks like this "image name in tag" feature is dead code not used since at least 2 years. I'm opening another PR to delete this code and I'll help you rebase here.

TL;DR; should be LESS code for you !

Ref. https://github.com/jenkins-infra/pipeline-library/pull/925